### PR TITLE
Enable `external-helpers` by default in tests

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This PR updates our test runner so that it always uses the `external-helpers` plugin, unless it's disabled with `"externalHelpers": false` in `options.json`.

This has the following advantages:
- We reduce the "visual noise" in our `output.js` files, helping reviewers focusing on the important output
- When an helper changes, we don't get a ton of updated fixtures just because the internal code of the helper was updated (example: https://github.com/babel/babel/pull/8461)
- There will be a single place where we'll need to update the `helperVersion` option, rather than hundreds of them.

This PR is inspired from https://github.com/babel/babel/pull/11285#discussion_r394753238. <s>but I can't find it anymore </s>:sweat_smile:

### CODE REVIEW

**Comment by mwwm1** [comment]
م

**Review by JLHwung — APPROVED** [review]
The new option should also be reflected in #12619.

**Review by hzoo — APPROVED** [review]
💯💯 amazing idea, makes a lot of sense! I'm just wondering where we want to document this like in contributing that this is the default or in some readme's in the test runner?

like a section in https://github.com/babel/babel/blob/main/CONTRIBUTING.md#babelplugin-x? We don't have that many docs on helpers in general really or explaining that we need them

**Comment by nicolo-ribaudo** [comment]
Good idea, I added a small section to `CONTRIBUTING.md`.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -213,24 +213,43 @@ expect(2 ** 3).toBe(8);
...
+  ```jsonc
+  // options.json example
+  {
+    "minNodeVersion": "5.0.0"
+  }
+  ```
+
+- **`externalHelpers`** (boolean)
+
+  By default, all the tests run with the `@babel/plugin-external-helpers` enabled. You can disable this behavior with
```
Comment (JLHwung) [inline CONTRIBUTING.md:245]:
```suggestion
  By default, all the tests run with the [`@babel/plugin-external-helpers`](https://babel.dev/docs/en/babel-plugin-external-helpers) enabled. You can disable this behavior with
```
though we don't have good docs there either.

**Review by JLHwung — APPROVED** [review]
The new `options.json` docs looks good to me!

**Pull request merged**